lib/Service/MailManager.php

```diff
@@ -282,6 +282,37 @@ public function markFolderAsRead(Account $account, string $folderId): void {
 		$this->messageMapper->markAllRead($client, $folderId);
 	}
 
+	/**
+	 * @param Account $account
+	 * @param string $mailbox
+	 * @param bool $subscribed
+	 *
+	 * @return bool
+	 * @throws ClientException
+	 * @throws ServiceException
+	 *
+	 */
+
+	public function subscribeFolder(Account $account, string $mailbox, bool $subscribed): bool {
+		$client = $this->imapClientFactory->getClient($account);
+		try {
+			$mb = $this->mailboxMapper->find($account, $mailbox);
+		} catch (DoesNotExistException $e) {
+			throw new ClientException("Mailbox $mailbox does not exist", 0, $e);
+		}
+		try {
+			$client->subscribeMailbox($mailbox, $subscribed);
+		} catch (Horde_Imap_Client_Exception $e) {
+			throw new ServiceException("Could not set subscription status for mailbox $mailbox on IMAP: " . $e->getMessage(), $e->getCode(), $e);
+		}
+		$folders = $this->folderMapper->getFolders($account, $client, $mailbox);
+		if (count($folders) !== 1) {
+			throw new ServiceException("Could not identify IMAP Folder: " . $e->getMessage(), $e->getCode(), $e);
```

Comment:
`$e` is not defined. There is no previous exception.

Reply:
Ok. I still missed that. Sorry...

Reply:
don't worry :)